first_hour_on_vwb/creating_a_data_collection.ipynb

```diff
@@ -0,0 +1,709 @@
+    "Run the cell below to create a widget, then populate the widget's input fields and click the button to create a version and publish it.\n",
+    "\n",
+    "Widget input parameters include:\n",
+    "- `workspace_id`: Must match the workspace ID of the workspace created above.\n",
+    "- `version`: Must be a string. This field is useful in particular if future releases of the data in this data collection are planned. It's suggested that the same versioning strategy be used for all releases of a data collection to make it transparent for users (e.g., if the first release is version `1.0`, the next release should be `2.0`, not `Version 2`).\n",
+    "- `description`: Must be a string. This field should describe this particular version of your data collection and is displayed in the data catalog.\n",
+    "\n",
+    "The output should resemble:\n",
+    "```\n",
+    "Workspace properties successfully updated.\n",
+    "ID:                <STUDY_NAME>_<YYMMDD>_dc_ws\n",
+    "Name:               <STUDY_NAME>-Data-<YYMMDD>\n",
+    "Description:       <DESCRIPTION>\n",
+    "Cloud Platform:    GCP\n",
+    "Google project:    <GOOGLE_PROJECT_ID>\n",
+    "Cloud console:     https://console.cloud.google.com/home/dashboard?project=<GOOGLE_PROJECT_ID>\n",
+    "Properties:\n",
+    "  terra-workspace-short-description: descriptive content\n",
+    "  terra-workspace-version: 1.0\n",
+    "  terra-type: data-collection\n",
+    "Created:           YYYY-MM-DD\n",
+    "Last updated:      YYYY-MM-DD\n",
+    "# Resources:       Workspace properties successfully updated.\n",
+    "ID:                <STUDY_NAME>_<YYMMDD>_dc_ws\n",
+    "Name:              <STUDY_NAME> Data <YYMMDD>\n",
+    "Description:       <DESCRIPTION>\n",
+    "Cloud Platform:    GCP\n",
+    "Google project:    <GOOGLE_PROJECT_ID>\n",
+    "Cloud console:     https://console.cloud.google.com/home/dashboard?project=<GOOGLE_PROJECT_ID>\n",
+    "Properties:\n",
+    "  terra-workspace-short-description: <SHORT_DESCRIPTION>\n",
+    "  terra-workspace-version: <VERSION>\n",
+    "  terra-type: data-collection\n",
+    "Created:           YYYY-MM-DD\n",
+    "Last updated:      YYYY-MM-DD\n",
+    "# Resources:       0\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "class PublishVersionWidget(object):\n",
+    "    def __init__(self, prev_widget):\n",
+    "        self.label = widgets.Label(value = 'Please provide appropriate values in the input boxes.')\n",
+    "        self.data_collection_ids = self.get_data_collections()\n",
+    "        if prev_widget and prev_widget.get_workspace_id() != '':\n",
+    "            self.input_workspace_id = wu.DropdownInputWidget(self.data_collection_ids,prev_widget.get_workspace_id(),\"Workspace ID:\").get()\n",
```

Comment:
I got an error at this point saying `ValueError: 'willyn-workspace-dc-test' not in array`.

It turns out this wasn't your fault, but `terra workspace list` has a default limit of 30. `willyn-workspace-dc-test` happened to be my 31st workspace. Might be worth adding `--limit=100` or something to the `terra workspace list` call.

Reply:
Now that I've refactored this (per our Slack convo yesterday), this comment is obsolete.